src/main/java/com/microsoft/jenkins/azuread/AzureSecurityRealm.java

```diff
@@ -517,6 +518,11 @@ public SecurityComponents createSecurityComponents() {
                 } catch (GraphServiceException e) {
                     if (e.getResponseCode() == NOT_FOUND) {
                         return null;
+                    } else if (e.getResponseCode() == BAD_REQUEST) {
+                        LOGGER.log(Level.WARNING, "Failed to lookup user with userid '" + userId + "'."
+                                + " Enable 'Fine' Logging for more information.");
+                        LOGGER.log(Level.FINE, e.getMessage());
```

Comment:
if we're logging it at fine just log the whole exception?

Reply:
Sure.